panic while using non X11 tty

I am receiving the following error while using a go project over a non X11 tty.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x58 pc=0x43a26a]

goroutine 1 [running]:
runtime.panic(0x529780, 0x6b94a8)
    /usr/local/go/src/pkg/runtime/panic.c:266 +0xb6
github.com/proxypoke/i3ipc.(*IPCSocket).Raw(0xc21001d360, 0x2, 0xc21000a430, 0xd, 0x0, ...)
    $GOPATH/src/github.com/proxypoke/i3ipc/socket.go:171 +0x4ba
github.com/proxypoke/i3ipc.(*IPCSocket).subscribe(0xc21001d360, 0x7fe600000000, 0x0, 0x0)
    $GOPATH/src/github.com/proxypoke/i3ipc/subscribe.go:69 +0xc9
github.com/proxypoke/i3ipc.init·1()
    $GOPATH/src/github.com/proxypoke/i3ipc/subscribe.go:148 +0x11f
github.com/proxypoke/i3ipc.init()
    $GOPATH/src/github.com/proxypoke/i3ipc/workspaces.go:45 +0x60
main.init()
    $GOPATH/src/github.com/Avinash-Bhat/i3go/main.go:79 +0x47
```

is this a i3ipc issue or i3wm issue or a go issue like [this](http://code.google.com/p/go/issues/detail?id=5193)?

the project in question is over [here](https://github.com/Avinash-Bhat/i3go)

--- a/subscribe.go
+++ b/subscribe.go
@@ -145,6 +145,9 @@ func init() {
 	var ev EventType
 	for ; ev < eventmax; ev++ {
 		sock, err := GetIPCSocket()
+		if err != nil {
+			log.Fatalf("Can't get i3 socket. Please make sure i3 is running. %v.", err)
+		}
 		err = sock.subscribe(ev)
 		if err != nil {
 			log.Fatalf("Can't subscribe: %v", err)

